src/top.rs

```diff
@@ -242,7 +262,10 @@ impl GeckoMedia {
     /// finished playing media. Returns Err() if any GeckoMedia or Player
     /// objects remain undropped.
     pub fn shutdown() -> Result<(), ()> {
-        let mut sender = SENDER.lock().unwrap();
+        if unsafe { SENDER }.is_null() {
```

Comment:
How do you guarantee that something doesn't call GeckoMedia::get() on another thread right after the calling thread bails out here early after assuming gecko media wasn't started? The other thread would start up the media stack after the calling thread exited thinking it was shutdown.

Reply:
I am going to change this code.